### PR TITLE
(RENEW) Add database to bot

### PR DESCRIPTION
**Pull request by blaumeise20**
Finally we have a database in our bot! Now we can use it in the modules to store data and create cool commands.

(RENEW because of problems with the original PR)

### CODE REVIEW

File: DiscordBot.Database/DatabaseHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            byte[] primaryKeyHash;
+            if (props.PrimaryKeys.Count > 0)
+            {
+                StringBuilder builder = new StringBuilder();
+                foreach (var key in props.PrimaryKeys)
+                {
+                    builder.Append(key.GetMethod.Invoke(storeObject, null)?.ToString() ?? "null");
+                }
+
+                primaryKeyHash = MD5.Create().ComputeHash(Encoding.UTF8.GetBytes(builder.ToString()));
```
Comment (JKamsker) [inline DiscordBot.Database/DatabaseHelpers.cs:42]: CSP Still has to be disposed

File: DiscordBot.Database/DatabaseHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                storeObject.PartKey = Convert.ToBase64String(primaryKeyHash, 0, 5);
+                storeObject.Id = Convert.ToBase64String(primaryKeyHash, 5, primaryKeyHash.Length - 5);
+            }
+        }
+
+        private static readonly Dictionary<Type, ObjectStoreProperties> _knownTypes = new Dictionary<Type, ObjectStoreProperties>();
+        public static ObjectStoreProperties GetProperties<T>() where T : class
+        {
+            if (!_knownTypes.ContainsKey(typeof(T)))
```
Comment (JKamsker) [inline DiscordBot.Database/DatabaseHelpers.cs:63]: TrygetValue

File: DiscordBot.Database/DatabaseService.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿using DiscordBot.Domain.Configuration;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Options;
+
+namespace DiscordBot.Domain.Database
```
Comment (JKamsker) [inline DiscordBot.Database/DatabaseService.cs:5]: https://github.com/coderdojo-linz/DiscordBot-DotNet/pull/56#discussion_r452138728

Comment (blaumeise20) [reply DiscordBot.Database/DatabaseService.cs:5]: But what's this: https://github.com/coderdojo-linz/DiscordBot-DotNet/blob/master/DiscordBot.CoderDojoInfoModule/CoderDojoInfoModule.cs#L11 ? There is also code.

Comment (JKamsker) [reply DiscordBot.Database/DatabaseService.cs:5]: I dont know. Must be a mistake

**Pull request merged**